SwiftUI_DZ/SwiftUI_DZ/ShopView/ShopView.swift

```diff
@@ -0,0 +1,181 @@
+            .frame(width: Constants.sizeImage, height: Constants.sizeImage)
+            .scaledToFill()
+    }
+    
+    private var aboutProductImageView: some View {
+        Image(Constants.emptyString)
+            .resizable()
+            .aspectRatio(contentMode: .fill)
+            .frame(width: Constants.aboutProductImageSizeValue.0, height: Constants.aboutProductImageSizeValue.1)
+            .clipShape(Rectangle())
+            .overlay {
+                Rectangle()
+                    .foregroundColor(.white)
+            }
+            .cornerRadius(10)
+            .shadow(radius: 7)
+    }
+    
+    private var costProductHStackView: some View {
+        VStack {
+            Text(shopViewModel.product.name)
+                .font(.system(size: 24))
+                .fontWeight(.bold)
+            Image(Constants.emptyString)
+                .resizable()
+                .frame(width: 80, height: 3)
+                .background(.yellow)
+            Text("\(shopViewModel.product.cost)\(Constants.currencyString)")
+                .font(.system(size: 35))
+                .fontWeight(.heavy)
+        }
+    }
+    
+    private var recommendBuyHStackView: some View {
+        VStack(spacing: 20) {
+            HStack {
+                Spacer().frame(width: Constants.spacerWidthValue)
+                Image(Constants.heartImageName)
+                    .resizable()
+                    .frame(width: 30, height: 30)
+            }
+            Button {
+                shopViewModel.isBuy = true
+            } label: {
+                Text(Constants.buyNowString)
+                    .font(.system(size: Constants.sizeFontDouble, weight: .heavy, design: .default))
+                    .foregroundColor(.white)
+                    .padding()
+                    .frame(maxWidth: 100)
+                    .background(Color.red)
+            }
+            .cornerRadius(Constants.getStartedCornerRadiusValue)
+            .frame(width: Constants.buyButtonWidthValue)
+        }
+    }
+    
+    private var descriptionTitleHStackView: some View {
+        HStack {
+            Image(Constants.bagImageName)
+                .resizable()
+                .frame(width: 50, height: 50)
+            Text(Constants.descriptionString)
+                .font(.system(size: 20))
+                .fontWeight(.heavy)
+            Spacer()
+        }
+        .padding()
+    }
+    
+    private var descriptionVStackView: some View {
+        VStack {
+            TextEditor(text: $shopViewModel.text)
+                .focused($idDescriptionFocused)
+                .padding()
+                .foregroundColor(.black)
+                .font(.custom(Constants.fontNameSting, size: 20, relativeTo: .body))
+                .frame(width: 400, height: 120)
+                .onChange(of: shopViewModel.text) { text in
+                        shopViewModel.totalChars = text.count
+                    if shopViewModel.totalChars <= 150 {
```

Comment:
```suggestion
                    if shopViewModel.totalChars <= 150 {
                    
```
Думаю вот это все таки магическое число